modules/core/src/v2/coins/utxo/recovery/backupKeyRecovery.ts

```diff
@@ -253,43 +258,24 @@ export async function backupKeyRecovery(
     RootWalletKeys.defaultPrefix,
   ]);
 
-  const queries: Promise<WalletUnspent[]>[] = [];
-
-  _.forEach(Object.keys(Codes.UnspentTypeTcomb.meta.map), function (addressType) {
-    // If we aren't ignoring the address type, we derive the public key and construct the query for the external and
-    // internal indices
-    if (!_.includes(params.ignoreAddressTypes, addressType)) {
-      if (addressType === Codes.UnspentTypeTcomb('p2shP2wsh') && !coin.supportsP2shP2wsh()) {
-        // P2shP2wsh is not supported for this coin so we skip this unspent type.
-        return;
-      }
-
-      if (addressType === Codes.UnspentTypeTcomb('p2wsh') && !coin.supportsP2wsh()) {
-        // P2wsh is not supported for this coin so we skip this unspent type.
-        return;
-      }
-
-      if (addressType === Codes.UnspentTypeTcomb('p2tr') && !coin.supportsP2tr()) {
-        // P2tr is not supported for this coin so we skip this unspent type.
-        return;
-      }
-
-      let codes;
-      try {
-        codes = Codes.forType(Codes.UnspentTypeTcomb(addressType) as any);
-      } catch (e) {
-        // The unspent type is not supported by bitgo so attempting to get its chain codes throws. Catch that error
-        // and continue.
-        return;
-      }
-      queries.push(queryBlockchainUnspentsPath(coin, params, walletKeys, codes.external));
-      queries.push(queryBlockchainUnspentsPath(coin, params, walletKeys, codes.internal));
-    }
-  });
+  const unspents: WalletUnspent[] = _.flatten(
+    await Promise.all(
+      outputScripts.scriptTypes2Of3
+        .filter(
+          (addressType) => coin.supportsAddressType(addressType) && !params.ignoreAddressTypes?.includes(addressType)
+        )
+        .reduce(
+          (queries, addressType) => [
+            ...queries,
+            queryBlockchainUnspentsPath(coin, params, walletKeys, getExternalChainCode(addressType)),
+            queryBlockchainUnspentsPath(coin, params, walletKeys, getInternalChainCode(addressType)),
+          ],
+          [] as Promise<WalletUnspent[]>[]
+        )
```

Comment:
```suggestion
        .map((addressType) => ([
          queryBlockchainUnspentsPath(coin, params, walletKeys, getExternalChainCode(addressType)),
          queryBlockchainUnspentsPath(coin, params, walletKeys, getInternalChainCode(addressType)),
        ]))
        .flat()
```

Reply:
sadly not available with the current compilation target set in `tsconfig.json` (we need es2019 and have es2017)

I didn't want to blow up this PR even more

Reply:
https://github.com/BitGo/BitGoJS/pull/1774